static/src/javascripts/projects/common/modules/email/email-article.js

```diff
@@ -0,0 +1,93 @@
+define([
+    'common/utils/$',
+    'bean',
+    'bonzo',
+    'common/modules/identity/api',
+    'fastdom',
+    'common/modules/email/email',
+    'common/utils/detect',
+    'lodash/collections/contains',
+    'common/utils/config',
+    'lodash/collections/every',
+    'lodash/collections/find'
+], function (
+    $,
+    bean,
+    bonzo,
+    Id,
+    fastdom,
+    email,
+    detect,
+    contains,
+    config,
+    every,
+    find
+) {
+
+    var listConfigs = {
+            nhs: {
+                listId: '3573',
+                canRun: 'nhs',
+                campaignCode: 'NHS_in_article',
+                headline: 'Interested in the NHS?',
+                description: 'Sign up to email updates related to the Guardian\'s coverage of the NHS, including daily updates as the project develops',
+                successHeadline: 'Thank you for signing up to our NHS email updates',
+                successDescription: 'You\'ll receive daily updates in your inbox'
+            }
+        },
+        $articleBody,
+        emailInserted = false,
+        isParagraph = function ($el) {
+            return $el.nodeName && $el.nodeName === 'P';
+        },
+        listCanRun = function (listConfig) {
+            if (listConfig.canRun && canRunList[listConfig.canRun]()) {
+                return listConfig;
+            }
+        },
+        addListToPage = function (listConfig) {
+            if (listConfig) {
+                var iframe = bonzo.create('<iframe src="/email/form/article/' + listConfig.listId + '" height="218px" data-form-title="' + listConfig.headline + '" data-form-description="' + listConfig.description + '" data-form-campaign-code="' + listConfig.campaignCode + '" scrolling="no" seamless frameborder="0" class="iframed--overflow-hidden email-sub__iframe js-email-sub__iframe js-email-sub__iframe--article"' + ' data-form-success-headline="' + listConfig.successHeadline + '" data-form-success-desc="' + listConfig.successDescription + '"></iframe>')[0];
+
+                bean.on(iframe, 'load', function () {
+                    email.init(iframe);
+                });
+
+                fastdom.write(function () {
+                    $(iframe).appendTo($articleBody);
```

Comment:
maybe do the dom lookup outside the callback, to minimise the activity tied to the frame?